### PR TITLE
Product catalog rendering and transactional system

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-495, https://github.com/ibexa/product-catalog/pull/342
| Versions      | 4.1, master

Note: for 4.2 replace the custom controller for product list rendering with a Query Type, ref. https://github.com/ibexa/product-catalog/pull/480

Also: removes the "Embed product" page, which is irrelevant now that overriding templates for product is the same as with regular content.

### CODE REVIEW

File: docs/guide/catalog/set_up_products.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Set up product for purchasing
```
Comment (DominikaK) [inline docs/guide/catalog/set_up_products.md:1]: For doc reviewers: not quite satisfied with the title, happy to hear better alternatives

File: docs/guide/catalog/set_up_products.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    - germany
+                    - norway
+```
+
+### Configuring other regions and currencies
+
+By default, the system always uses the first currency and the first region configured.
+
+To implement a different logic, for example a switcher for preferred currencies and regions,
+you need to use `CurrencyResolverInterface` and `RegionResolverInterface` in you customization.
```
Comment (adamwojs) [inline docs/guide/catalog/set_up_products.md:35]:
```suggestion
To implement a different logic, for example a switcher for preferred currencies and regions,
you need to use `CurrencyResolverInterface` and `RegionResolverInterface` in you customization.
```

`\Ibexa\Contracts\ProductCatalog\Events\CurrencyResolveEvent` and `\Ibexa\Contracts\ProductCatalog\Events\RegionResolveEvent` should be subscribed in order to customize this logic. 

Mentioned services are meet to be used to read current currency / region.

File: code_samples/front/shop/list_products/src/Controller/ProductListController.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Controller;
+
+use Ibexa\Contracts\ProductCatalog\Values\Product\ProductQuery;
+use Ibexa\ProductCatalog\Local\Repository\ProductService;
+use Ibexa\Core\MVC\Symfony\View\View;
+
+class ProductListController
```
Comment (adamwojs) [inline code_samples/front/shop/list_products/src/Controller/ProductListController.php:9]:
```suggestion
final class ProductListController
```

File: code_samples/front/shop/list_products/src/Controller/ProductListController.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Controller;
+
+use Ibexa\Contracts\ProductCatalog\Values\Product\ProductQuery;
+use Ibexa\ProductCatalog\Local\Repository\ProductService;
```
Comment (adamwojs) [inline code_samples/front/shop/list_products/src/Controller/ProductListController.php:6]: \\Ibexa\\Contracts\\ProductCatalog\\ProductServiceInterface should be used here.

File: code_samples/front/shop/list_products/src/Controller/ProductListController.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+use Ibexa\Core\MVC\Symfony\View\View;
+
+class ProductListController
+{
+    private $productService;
```
Comment (adamwojs) [inline code_samples/front/shop/list_products/src/Controller/ProductListController.php:11]:
```suggestion
    private ProductServiceInterface $productService;
```

File: code_samples/front/shop/list_products/src/Controller/ProductListController.php [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+    public function __construct(ProductService $productService)
```
Comment (adamwojs) [inline code_samples/front/shop/list_products/src/Controller/ProductListController.php:13]:
```suggestion
    public function __construct(ProductServiceInterface $productService)
```

File: code_samples/front/shop/list_products/src/Controller/ProductListController.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        $this->productService = $productService;
+    }
+
+    public function showProductsAction(View $view)
```
Comment (adamwojs) [inline code_samples/front/shop/list_products/src/Controller/ProductListController.php:18]:
```suggestion
    public function showProductsAction(View $view): View
```

**Review by adamwojs — CHANGES_REQUESTED** [review]

File: code_samples/front/shop/list_products/src/Controller/ProductListController.php (hunk 4 above)
Comment (DominikaK) [inline code_samples/front/shop/list_products/src/Controller/ProductListController.php:6]:
```suggestion
use Ibexa\Contracts\ProductCatalog\ProductServiceInterface;
```

File: docs/guide/catalog/set_up_products.md (hunk 2 above)
Comment (DominikaK) [inline docs/guide/catalog/set_up_products.md:35]:
```suggestion
To implement a different logic, for example a switcher for preferred currencies and regions,
you need to subscribe to `Ibexa\Contracts\ProductCatalog\Events\CurrencyResolveEvent`
and `Ibexa\Contracts\ProductCatalog\Events\RegionResolveEvent` in your customization.
```

File: docs/guide/catalog/set_up_products.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
+# Set up product for purchasing
+
+To enable purchasing from the catalog, the following configuration is required:
+
+- at least [one region and one currency for the shop](#region-and-currency)
+- [VAT rates for the product type](#vat-rates)
+- at least [one price for the product](#product-price)
+- [availability with positive or infinite stock for the product](#product-availability)
```
Comment (juskora) [inline docs/guide/catalog/set_up_products.md:8]: How about adding direct links to Product price and Product availability. Currently there is one more click to get to the point, maybe we could reduce it to only one :)

**Review by juskora — CHANGES_REQUESTED** [review]

**Review by juskora — APPROVED** [review]

**Pull request merged**